### ISSUE TITLE
[Feature Request] Add a HUD Toggle State that shows Map and Mission Progress at the same time

### ISSUE THREAD

**Issue by TheGershon** [vitor251093/KHMelonMix#206]
Displaying the current World is functionally useless, it would be much more convenient if players could see both Mission Progress and the Map at the same time as it's incredibly annoying to have to constantly switch states if you want to keep your eyes on both. Current World could maybe be relegated to the pause menu, or its own HUD State.

Here's a mockup of what it would look like:
![Untitled](https://github.com/user-attachments/assets/9f02e0b9-96b8-43b8-be5d-7aa16d1156c3)

**Comment by sandwichwater** [comment]
the main point of the second toggle state is to show the mission target , the current world being shown is just a side effect for things like recon or heart collection just due to the nature of how the current objective works . that being said , maybe we could add a fourth state

**Comment by TheGershon** [comment]
Right, I forgot that it's Target as well. Regardless, it's very rarely actually useful imo. Would much prefer the map, but then that sacrifices the mission gauge. Thus my dilemma.

**Comment by vitor251093** [comment]
I can think of three options:

- Replacing it with the map when there is no Target;

- Creating an option to always show the mission gauge;

- Creating an alternative mode (which is technically a fourth state, but enabled by options) called Auto, which picks automatically what should be shown. Is there Mission Information? Show. Mission Gauge? Always there. Is there a target? Show the target. No target? Show the map.

**Comment by TheGershon** [comment]
Personally, I'd prefer the second option. Have the mission gauge and the map/target be on separate toggles.

**Comment by EpicGamerNaeNae** [comment]
Another alternative could be having a menu that allows users to customize which elements appear on each HUD state.
The target box would just have to be moved to not overlap with the map, and you could have a state woth every UI element or as few as the user wants.
This could maybe also apply to other HUD elements like health bars and the jeart counter, and potentially also allow to move them around (though that might not be worth the effort)

**Comment by vitor251093** [comment]
> Another alternative could be having a menu that allows users to customize which elements appear on each HUD state. 

Idk, that would require a whole menu section dedicated to that, and would also lead to the need to add and remove states, which would make the new game settings way more complex on the get go. We can hold that for a future release though.

> The target box would just have to be moved to not overlap with the map
It's possible, but only if the HUD elements size as a whole decreases, in which case it could be next to the map (a HUD size slider is already on the plans though, and the plugin is already ready for it). If the HUD size is kept, this would lead to a very clustered HUD.

> and you could have a state woth every UI element or as few as the user wants
The clustered HUD is also a problem in that case, but I guess that if the user wants to have them, this would be an option.

> This could maybe also apply to other HUD elements like health bars and the jeart counter
Ironically, that's the easiest part.

> and potentially also allow to move them around (though that might not be worth the effort)

Yeah, definitely not xD

They would also need to be able to control the size of the elements, and to define to which corner of the screen their position should be relative to (not only corners in fact, because the mission gauge position is relative to the bottom center).

This would also make the fragment shader WAY more complex, leading to lower frame rates for everyone.

**Comment by vitor251093** [comment]
> Personally, I'd prefer the second option. Have the mission gauge and the map/target be on separate toggles.

We can do that before reaching the more complex customizable HUD states.
